### DISCUSSION TITLE
Integration into Swagger

### DISCUSSION THREAD

**Discussion by stefannikolei** [SixLabors/ImageSharp.Web#128] [Ideas]
Wouldn't it be neat to show the people who use this middleware what functionality they can use?

If we implement iApiDescriptionProvider the Middleware can be found in the swagger.

At the moment i am experimenting with this approach locally. It looks promising. 
Only struggling with the Datatypes and the Enums. But the Main description is already working.

What do you think?

**Comment by JimBobSquarePants** [comment]
Hmmm... This something that has never occurred to me. 

I'm curious to see what you have come up with. It would have to ensure there are no breaking changes.

**Reply by stefannikolei** [reply]
No that is non breaking.

There just needs to be an implementation of IApiDescriptionProvider. This needs to be added to DI and that all.

An example for a custom implementation can be found here: 

https://github.com/aspnet/AspLabs/blob/master/src/GrpcHttpApi/src/Microsoft.AspNetCore.Grpc.Swagger/Internal/GrpcHttpApiDescriptionProvider.cs

Could this be something that is interesting for this lib?

**Reply by JimBobSquarePants** [reply]
As long as it doesn't add any significant overhead to writing extensions and causes zero slowdown then I'm ok with it. 

Are you considering just the URL API? or would this factor into request URL sources?

**Reply by stefannikolei** [reply]
when putting much effort in it the providers must be checked for the path they handle so we can show them in swagger.

This doesn't have any runtime overhead. The provider is once executed when the application generates the ApiDescription

**Reply by JimBobSquarePants** [reply]
You've definitely piqued my curiosity. If you fancy opening a draft PR we can work to implement it together.

**Reply by stefannikolei** [reply]
https://github.com/SixLabors/ImageSharp.Web/pull/129 Here is the initial draft
